component-framework/TS_PropertySetTableControl/TSPropertySetTableControl/index.ts

```diff
@@ -0,0 +1,355 @@
+		}
+
+	/**
+	 * Called when any value in the property bag has changed. This includes field values, data-sets, global values such as container height and width, offline status, control metadata values such as label, visible, etc.
+	 * @param context The entire property bag available to control via Context Object; It contains values as set up by the customizer mapped to names defined in the manifest, as well as utility functions
+	 */
+	public updateView(context: ComponentFramework.Context<IInputs>): void {
+		this.contextObj = context;
+		this.toggleLoadMoreButtonWhenNeeded(context.parameters.sampleDataSet);
+		this.toggleLoadPreviousButtonWhenNeeded(context.parameters.sampleDataSet);
+		
+		if (!context.parameters.sampleDataSet.loading) {
+
+			// Get sorted columns on View
+			let columnsOnView = this.getSortedColumnsOnView(context);
+			this.columns = columnsOnView;
+			if (!columnsOnView || columnsOnView.length === 0) {
+				return;
+			}
+
+			let columnWidthDistribution = this.getColumnWidthDistribution(context, columnsOnView);
+
+			while (this.dataTable.firstChild) {
+				this.dataTable.removeChild(this.dataTable.firstChild);
+			}
+
+			this.dataTable.appendChild(this.createTableHeader(columnsOnView, columnWidthDistribution));
+			this.dataTable.appendChild(this.createTableBody(columnsOnView, columnWidthDistribution, context.parameters.sampleDataSet));
+
+			this.dataTable.parentElement!.style.height = (context.mode.allocatedHeight - 50) + "px";
+		}
+	}
+
+	/**
+	 * It is called by the framework prior to a control receiving new data.
+	 * @returns an object based on nomenclature defined in manifest, expecting object[s] for property marked as “bound” or “output”
+	 */
+	public getOutputs(): IOutputs {
+		return {};
+	}
+
+	/**
+		 * Called when the control is to be removed from the DOM tree. Controls should use this call for cleanup.
+	 * i.e. cancelling any pending remote calls, removing listeners, etc.
+	 */
+	public destroy(): void {
+	}
+
+	/**
+	 * Get sorted columns on view
+	 * @param context
+	 * @return sorted columns object on View
+	 */
+	private getSortedColumnsOnView(context: ComponentFramework.Context<IInputs>): DataSetInterfaces.Column[] {
+		if (!context.parameters.sampleDataSet.columns) {
+			return [];
+		}
+
+		let columns = context.parameters.sampleDataSet.columns;
+
+		return columns;
+	}
+
+	/**
+	 * Get column width distribution
+	 * @param context context object of this cycle
+	 * @param columnsOnView columns array on the configured view
+	 * @returns column width distribution
+	 */
+	private getColumnWidthDistribution(context: ComponentFramework.Context<IInputs>, columnsOnView: DataSetInterfaces.Column[]): string[] {
+
+		let widthDistribution: string[] = [];
+
+		// Considering need to remove border & padding length
+		let totalWidth: number = context.mode.allocatedWidth;
+		let widthSum = 0;
+
+		columnsOnView.forEach(function (columnItem) {
+			widthSum += columnItem.visualSizeFactor;
```

Comment:
What is visualSizeFactor ? does it work for excel/sql data source ? Please add some comments on whats being done in this sample. I know this is missing in the main one too so loop in Anqi, he can do the honors :)